### DISCUSSION TITLE
Is it possible to include a json file containing a configuration step within a recipe?

### DISCUSSION THREAD

**Discussion by DotCat1985** [OrchardCMS/OrchardCore#8337] [Q&A]
Is it possible to include a json file containing a configuration step within a recipe?

**Comment by hishamco** [comment]
AFAIK there's ability to access the configuration from `appsettings.json` from with in the recipe using `[js: configuration()]`, have a look to https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore.Themes/TheBlogTheme/Recipes/blog.recipe.json

**Reply by DotCat1985** [reply]
OK. With `[js: configuration()]` I could include settings from `appsettings.json`.
But, how can I include a json file placed in a theme project and containing a step configuration?

**Comment by jtkech** [comment]
What you can do is to execute a recipe inside a recipe, so an inner recipe

To do this, there is a specific recipe step whose name is just `Recipes`

But not sure what you mean with `configuration step`

**Reply by DotCat1985** [reply]
I would like to include for instance the following json within the recipe:

```
{
  "name": "WorkflowType",
  "data": [
    {
	"WorkflowTypeId": "[js:uuid()]",
	"Name": "Workflow name",
	"IsEnabled": true,
	"IsSingleton": false,
	"DeleteFinishedWorkflows": false,
	"Activities": [
           ...
	]
    }
  ]
}
```

**Reply by jtkech** [reply]
I think you mean `variables` in place of `data`

As i remember, parent variables can be re-used in an inner recipe, but not the opposite, a parent recipe using variables set by an inner recipe, but not sure, so it is worth to try it ;)

**Reply by DotCat1985** [reply]
Ok, thank you :)
